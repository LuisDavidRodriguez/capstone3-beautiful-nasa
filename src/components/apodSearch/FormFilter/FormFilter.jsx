import React, { useState } from 'react';

const FormFilter = () => {
  const [ state, setState ] = useState({});

  return (
    <form>
      <input>
      
      </input>
    </form>
  );
};

export default FormFilter;
