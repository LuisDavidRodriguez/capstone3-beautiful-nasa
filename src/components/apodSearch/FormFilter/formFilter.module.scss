@use '../../../styles/mixin';

.form {
  display: flex;
  flex-direction: column;

  @include mixin.small {
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }

  input {
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;       /* Remove default arrow */
    padding: 5px;
    border-radius: 5px;
    margin: 2px;
  }

  .selector {
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;       /* Remove default arrow */
    padding: 5px;
    border-radius: 5px;
    margin: 2px;
  }
}
