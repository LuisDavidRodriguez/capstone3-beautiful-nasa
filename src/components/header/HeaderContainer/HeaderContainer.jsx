import React from 'react';
import styles from './headerContainer.module.scss';
import VideoSimple from '../../ui/video/VideoSimple';

const sourceVideo = 'https://technology.nasa.gov/sites/default/files/2022-08/WebsiteLoopingvideo3.mp4';
const ariaLabel = 'NASA Technology Transfer Introduction Video';

const HeaderContainer = () => (
  <div className={styles.videoDiv}>
    <VideoSimple ariaLabel={ariaLabel} url={sourceVideo} />
    <Message />
  </div>
);

const Message = () => (
  <div className={styles.welcomeMessage}>
    <h3 className={styles.title}>Welcome to Beautiful-NASA</h3>
    <p
      className={styles.p1}
    >
      This application allows you to connect with NASA API&apos;s
    </p>
    <p className={styles.p1}>Enjoy the views and functionality</p>
    <p className={styles.p1}>
      Coded By
      {' '}
      <a
        href="https://github.com/LuisDavidRodriguez"
        target="_blank"
        rel="noreferrer"
        className={styles.a1}
      >
        Luis David Rodriguez Valades
      </a>
      {' '}
      powered by NASA
    </p>
  </div>
);

export default HeaderContainer;
