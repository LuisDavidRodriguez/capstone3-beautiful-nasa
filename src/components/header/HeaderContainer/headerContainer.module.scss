@use '../../../styles/functions' as f;
@use '../../../styles/variables' as v;
@use '../../../styles/mixin';

.videoDiv {
  // place relative to be able to move the text over the video
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 150px;
  min-width: 270px;
  width: 100%;
  max-height: 100vh;

  .welcomeMessage {
    position: absolute;
    width: 95%;
    max-width: 500px;
    background-color: transparentize($color: f.get-color('primary'), $amount: 0.25);
    padding: 1%;
    border-radius: 5px;
    box-shadow: 3px 3px 5px gray, -3px -3px 5px gray;
    display: flex;
    flex-direction: column;
    justify-content: center;

    .title {
      font-family: v.$fontFamilyLato;
      font-size: 1.2em;
      margin-bottom: 3px;
    }

    .p1,
    .p2,
    .p3 {
      font-family: v.$fontFamilyGillSans;
      font-weight: 600;
      font-size: 1em;
      margin-bottom: 3px;
    }

    .p2 {
      display: none;
    }

    .a1 {
      color: f.get-color("accentSecondary");
    }

    @include mixin.medium {
      padding: 2%;

      .p2 {
        display: block;
      }
    }
  }
}
