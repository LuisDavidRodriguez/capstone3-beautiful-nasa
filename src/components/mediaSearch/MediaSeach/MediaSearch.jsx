import React, { useState } from 'react';
import { PropTypes } from 'prop-types';
import styles from './mediaSearch.module.scss';

const MediaSearch = (props) => {
  const { reportInputs } = props;
  const [textInput, setTextInput] = useState('');
  const [selectInput, setSelectInput] = useState('All');

  const onChanegeManager = (e) => {
    const { name } = e.target;
    const { value } = e.target;

    if (name === 'textInput') {
      setTextInput(value);
      reportInputs({ text: value, mediaType: '' });
    }

    if (name === 'selectInput') {
      setSelectInput(value);
      reportInputs({ text: '', mediaType: value });
    }
  };

  const buttonClick = () => {
    setTextInput('');
    setSelectInput('All');
  };

  return (
    <form className={styles.form}>
      <input name="textInput" type="text" placeholder="Search for..." onChange={onChanegeManager} value={textInput} />
      <button type="button" onClick={buttonClick}>Search</button>
      <select value={selectInput} name="selectInput" onChange={onChanegeManager}>
        <option value="All">All</option>
        <option value="image">Image</option>
        <option value="video">Video</option>
      </select>
    </form>
  );
};

MediaSearch.defaultProps = {
};

MediaSearch.propTypes = {
  reportInputs: PropTypes.func.isRequired,
};

export default MediaSearch;
