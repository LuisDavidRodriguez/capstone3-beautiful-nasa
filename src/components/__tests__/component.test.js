import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import FormFilter from '../apodSearch/FormFilter/FormFilter';
import store from '../../redux/configStore';
import HeaderContainer from '../header/HeaderContainer/HeaderContainer';
import MediaSearch from '../mediaSearch/MediaSeach/MediaSearch';

describe('FormFilter components', () => {
  it('Render FormFitler', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <FormFilter
              reportInputs={() => {}}
              buttonHandler1={() => {}}
            />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('Header', () => {
  it('reder Header', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <HeaderContainer />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('Media search', () => {
  it('Media Search', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <MediaSearch
              reportInputs={() => {}}
            />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
