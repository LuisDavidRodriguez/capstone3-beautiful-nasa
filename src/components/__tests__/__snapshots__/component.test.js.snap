// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFilter components Render FormFitler 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
      >
        <input
          name="textInput"
          placeholder="search by text in this period"
          type="text"
          value=""
        />
        <input
          name="dateInput"
          type="date"
          value=""
        />
        <select
          class="selector"
          name="selectInput"
        >
          <option
            value="All"
          >
            All
          </option>
          <option
            value="image"
          >
            Image
          </option>
          <option
            value="video"
          >
            Video
          </option>
        </select>
        <button
          class="btn btn-outline-light"
          type="button"
        >
          default text1
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
    >
      <input
        name="textInput"
        placeholder="search by text in this period"
        type="text"
        value=""
      />
      <input
        name="dateInput"
        type="date"
        value=""
      />
      <select
        class="selector"
        name="selectInput"
      >
        <option
          value="All"
        >
          All
        </option>
        <option
          value="image"
        >
          Image
        </option>
        <option
          value="video"
        >
          Video
        </option>
      </select>
      <button
        class="btn btn-outline-light"
        type="button"
      >
        default text1
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header reder Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="videoDiv"
      >
        <video
          aria-label="NASA Technology Transfer Introduction Video"
          autoplay=""
          loop=""
          playsinline=""
          preload="true"
          width="100%"
        >
          <source
            src="https://technology.nasa.gov/sites/default/files/2022-08/WebsiteLoopingvideo3.mp4"
            type="video/mp4"
          />
          Sorry, your browser doesn't support videos.
        </video>
        <div
          class="welcomeMessage"
        >
          <h3
            class="title"
          >
            Welcome to Beautiful-NASA
          </h3>
          <p
            class="p1"
          >
            This application allows you to HIT 2 API's fournished by the NASA
          </p>
          <p
            class="p1"
          >
            Enjoyed the views and the functionality
          </p>
          <p
            class="p1"
          >
            Coded By
             
            <a
              class="a1"
              href="https://github.com/LuisDavidRodriguez"
              rel="noreferrer"
              target="_blank"
            >
              Luis David Rodriguez Valades
            </a>
             
            powered by the NASA
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="videoDiv"
    >
      <video
        aria-label="NASA Technology Transfer Introduction Video"
        autoplay=""
        loop=""
        playsinline=""
        preload="true"
        width="100%"
      >
        <source
          src="https://technology.nasa.gov/sites/default/files/2022-08/WebsiteLoopingvideo3.mp4"
          type="video/mp4"
        />
        Sorry, your browser doesn't support videos.
      </video>
      <div
        class="welcomeMessage"
      >
        <h3
          class="title"
        >
          Welcome to Beautiful-NASA
        </h3>
        <p
          class="p1"
        >
          This application allows you to HIT 2 API's fournished by the NASA
        </p>
        <p
          class="p1"
        >
          Enjoyed the views and the functionality
        </p>
        <p
          class="p1"
        >
          Coded By
           
          <a
            class="a1"
            href="https://github.com/LuisDavidRodriguez"
            rel="noreferrer"
            target="_blank"
          >
            Luis David Rodriguez Valades
          </a>
           
          powered by the NASA
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Media search Media Search 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
      >
        <input
          name="textInput"
          placeholder="Search for..."
          type="text"
          value=""
        />
        <button
          type="button"
        >
          Search
        </button>
        <select
          name="selectInput"
        >
          <option
            value="All"
          >
            All
          </option>
          <option
            value="image"
          >
            Image
          </option>
          <option
            value="video"
          >
            Video
          </option>
        </select>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
    >
      <input
        name="textInput"
        placeholder="Search for..."
        type="text"
        value=""
      />
      <button
        type="button"
      >
        Search
      </button>
      <select
        name="selectInput"
      >
        <option
          value="All"
        >
          All
        </option>
        <option
          value="image"
        >
          Image
        </option>
        <option
          value="video"
        >
          Video
        </option>
      </select>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
